src/components/Filter/FilterValue.tsx

```diff
@@ -0,0 +1,59 @@
+import { ComponentProps, forwardRef } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+import { Text, TextVariants } from '../Text';
+import Input from '../Input/Input';
+
+// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
+export interface FilterValueProps extends Omit<ComponentProps<'div'>, 'onChange'> {
+  // Text to be displayed as a label for the neighboring input
+  label: string;
+
+  // Type of input for this filter
+  inputType: 'checkbox' | 'radio';
+
+  // Should this filter be hidden, most common use will be for view all truncation
+  isHidden?: boolean;
+
+  // Whether or not this filter is disabled
+  disabled?: boolean;
+
+  // Function to trigger when the state of this filter is changed
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+/**
+ * ## Overview
+ *
+ * A label & input for filtering criteria
+ *
+ */
+const FilterValue = forwardRef<HTMLDivElement, FilterValueProps>(
+  ({ className, label, inputType, isHidden = false, onChange, disabled, ...props }, ref) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'FilterValue');
+    const disabledLabel = disabled ? 'disabled-label' : '';
+    return (
+      <>
+        {isHidden ? null : (
+          <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+            <Text variant={TextVariants.body2} className={`${baseClassName}__label ${disabledLabel}`}>
```

Comment:
we need a prefix here on disabledLabel to protect against css collisions


This should also be rendered as a label component with element="label" and ensure that the label is linked to the input for accessibility.  An example.  It might be better to actually use the `<Input label` instead of rendering your own text which hopefully should have the linkage built in and just apply custom styles to that label through scss.

<label for="tac">
  <input id="tac" type="checkbox" name="terms-and-conditions" />
  I agree to the Terms and Conditions.
</label>